dsl/pace/dsl/stencil.py

```diff
@@ -953,14 +1026,22 @@ def restrict_vertical(self, k_start=0, nk=None) -> "StencilFactory":
             grid_indexing=self.grid_indexing.restrict_vertical(k_start=k_start, nk=nk),
         )
 
+    def show_build_report(self, key: str = "build_time", **kwargs) -> str:
+        """Report all stencils built by this factory."""
+        return self.timing_collector.show_build_report(key, **kwargs)
+
+    def show_exec_report(self, key: str = "total_run_time", **kwargs) -> str:
+        """Report all stencils executed that were built by this factory."""
+        return self.timing_collector.show_exec_report(key, **kwargs)
```

Comment:
Please add tests of these functions. I think the code has a bug (key here is a descriptor, but in the lower levels it's a stencil class name), but it's easier to say if it does or doesn't if there's a test.

Reply:
Yes, I'll add a test. Looks correct to me:
```python
class TimingCollector:
...
    def show_exec_report(self, key: str = "total_run_time", **kwargs) -> str:
        return type(self)._show_report(self.exec_info, key, **kwargs)
```